src/DssDirectDepositHub.sol

```diff
@@ -0,0 +1,467 @@
+            // Otherwise we couldn't just simply take away the collateral from the End module as the next line will be doing.
+            vat.slip(ilk, address(end), -int256(amount));
+            vat.move(address(pool), vow, _mul(total, RAY));
+        }
+
+        emit Unwind(ilk, amount);
+    }
+
+    function exec(bytes32 ilk_) external {
+        D3MPoolLike pool = ilks[ilk_].pool;
+
+        uint256 availableAssets = pool.maxWithdraw();
+
+        if (vat.live() == 0) {
+            uint256 culled = ilks[ilk_].culled;
+            // MCD caged
+            require(end.debt() == 0, "DssDirectDepositHub/end-debt-already-set");
+            require(culled == 0, "DssDirectDepositHub/module-has-to-be-unculled-first");
+            _unwind(
+                ilk_,
+                pool,
+                type(uint256).max,
+                availableAssets,
+                Mode.MCD_CAGED
+            );
+        } else if (live == 0) {
+            uint256 culled = ilks[ilk_].culled;
+            // This module caged
+            _unwind(
+                ilk_,
+                pool,
+                type(uint256).max,
+                availableAssets,
+                culled == 1
+                ? Mode.MODULE_CULLED
+                : Mode.NORMAL
+            );
+        } else {
+            // Normal path
+            (uint256 totalAssets, uint256 targetAssets) = pool.calcSupplies(availableAssets);
+
+            if (targetAssets > totalAssets) {
+                _wind(ilk_, pool, targetAssets - totalAssets);
+            } else if (targetAssets < totalAssets) {
+                _unwind(
+                    ilk_,
+                    pool,
+                    totalAssets - targetAssets,
+                    availableAssets,
+                    Mode.NORMAL
+                );
+            }
+        }
+    }
+
+    // --- Collect Interest ---
+    function reap(bytes32 ilk_) external {
+        D3MPoolLike pool = ilks[ilk_].pool;
+
+        require(vat.live() == 1, "DssDirectDepositHub/no-reap-during-shutdown");
+        require(live == 1, "DssDirectDepositHub/no-reap-during-cage");
+
+        uint256 assetBalance = pool.assetBalance();
+        (, uint256 daiDebt) = vat.urns(ilk_, address(pool));
+        if (assetBalance > daiDebt) {
+            uint256 fees = assetBalance - daiDebt;
+            uint256 availableAssets = pool.maxWithdraw();
+            if (fees > availableAssets) {
+                fees = availableAssets;
+            }
+            pool.withdraw(fees);
+            daiJoin.join(address(this), fees);
+            vat.move(address(this), vow, _mul(RAY, fees));
+            emit Reap(ilk_, fees);
+        }
+    }
+
+    // --- Collect any rewards ---
+    function collect(bytes32 ilk_, address[] memory assets, uint256 amount) external returns (uint256 amt) {
+        D3MPoolLike pool = ilks[ilk_].pool;
+
+        amt = pool.collect(assets, amount);
+        emit Collect(ilk_, pool.king(), assets, amt);
```

Comment:
I'm a bit conflictive here,  I think it is nice the `king` is in the pool, but then not sure if I'd put it in the event. So without putting it in this event, the pool can implement whatever it wants, maybe multiple kings.

Reply:
Rewards are optional, platform-specific and usually pretty bespoke so I think we should move them entirely into the pool and have no concept of rewards, collect, king, etc in the Hub.

Reply:
that makes even more sense

Reply:
:+1: